### PR TITLE
Deprecate `@test_approx_eq` (part of #4615)

### PR DESCRIPTION
**Pull request by StefanKarpinski**
`@test_approx_eq_eps`, you're next!

### CODE REVIEW

**Comment by KristofferC** [comment]
Is there a non unicode way to do this?

**Comment by ararslan** [comment]
`@test isapprox(x, y)`

**Comment by cstjean** [comment]
They're not 100% equivalent yet, though. `isapprox` isn't informative.

```julia
julia> @test_approx_eq 2.1 1+1
ERROR: assertion failed: |2.1 - 1 + 1| <= 4.440892098500626e-12
  2.1 = 2.1
  1 + 1 = 2
  difference = 0.10000000000000009 > 4.440892098500626e-12
 in test_approx_eq(::Float64, ::Int64, ::Float64, ::String, ::String) at ./test.jl:863
 in test_approx_eq(::Float64, ::Int64, ::String, ::String) at ./test.jl:873

julia> @test isapprox(2.1, 1+1)
Test Failed
  Expression: isapprox(2.1,1 + 1)
ERROR: There was an error during testing
 in record(::Base.Test.FallbackTestSet, ::Base.Test.Fail) at ./test.jl:397
 in do_test(::Base.Test.Returned, ::Expr) at ./test.jl:281

julia> @test 2.1 ≈ 1+1
Test Failed
  Expression: 2.1 ≈ 1 + 1
   Evaluated: 2.1 isapprox 2
ERROR: There was an error during testing
 in record(::Base.Test.FallbackTestSet, ::Base.Test.Fail) at ./test.jl:397
 in do_test(::Base.Test.Returned, ::Expr) at ./test.jl:281
```

File: base/test.jl [hunk 1]
```diff
@@ -952,12 +952,15 @@ end
...
-small numerical errors.
+Deprecated. Test two floating point numbers `a` and `b` for equality taking into
+account small numerical errors.
 """
 macro test_approx_eq(a, b)
+    Base.depwarn(string("@test_approx_eq is deprecated, use `@test ", a, " ≈ ", b, "` instead"),
+                 Symbol("@test_approx_eq"))
     :(test_approx_eq($(esc(a)), $(esc(b)), $(string(a)), $(string(b))))
 end
+export @test_approx_eq
```
Comment (tkelman) [inline base/test.jl]: could we do this with an eval into the `Base.Test` module from `deprecated.jl` so it's in a place where we know we'll be deleting it?

**Comment by ararslan** [comment]
Fair point, @cstjean. We could check in `@test` whether the expression is a call to `isapprox`, but having to perform that check every time might make things slow and/or be overkill.

**Review by kshyatt — APPROVED** [review]

File: base/test.jl (hunk 1 above)
Comment (StefanKarpinski) [inline base/test.jl]: Eh, I'd rather just leave it here until we tear out the rest of this stuff.

**Comment by StefanKarpinski** [comment]
We display lots of test conditions specially, so that could certainly be arranged for `isapprox` tests.

File: doc/src/stdlib/test.md [hunk 2]
```diff
@@ -163,55 +163,6 @@ ERROR: Some tests did not pass: 3 passed, 1 failed, 0 errored, 0 broken.
  in finish at test.jl:362
 ```
 
-## Other Test Macros
-
-As calculations on floating-point values can be imprecise, you can perform approximate equality
-checks using either `@test a ≈ b` (where `≈`, typed via tab completion of `\approx`, is the
-[`isapprox()`](@ref) function) or use [`isapprox()`](@ref) directly.
```
Comment (tkelman) [inline doc/src/stdlib/test.md]: This first paragraph seems fine to leave in. Same with the examples that use `isapprox`, and the ` ```@docs ` line for `@inferred`

File: doc/src/stdlib/test.md [hunk 3]
```diff
@@ -169,46 +169,7 @@ As calculations on floating-point values can be imprecise, you can perform appro
...
-
-```julia
-julia> @test 1 ≈ 0.999999999
-
-julia> @test 1 ≈ 0.999999
-ERROR: test failed: 1 isapprox 0.999999
- in expression: 1 ≈ 0.999999
- in error at error.jl:21
- in default_handler at test.jl:30
- in do_test at test.jl:53
```
Comment (tkelman) [inline doc/src/stdlib/test.md]: these 2 were worth keeping

**Comment by giordano** [comment]
`@test_approx_eq` actually had a nice feature: `@test_approx_eq NaN NaN` didn't give an error.  I know there exists `isnan`, yet that feature was useful when comparing arrays made by a mixture of finite and NaN elements.  When doing tests, it makes sense to treat NaNs as equal to each other.

**Comment by StefanKarpinski** [comment]
I think I'd prefer to have that as a feature of isapprox than keep this weird extra testing API around.

File: doc/src/stdlib/test.md (hunk 3 above)
Comment (StefanKarpinski) [inline doc/src/stdlib/test.md]: Can you open an issue for it?

Comment (tkelman) [inline doc/src/stdlib/test.md]: about to get on a plane (aren't you, too?), I have a commit on a misc branch locally to put these back, can go into a PR next time I rearrange branches

**Comment by giordano** [comment]
I just wanted to point out a difference between `@test_approx_eq` and `isapprox`, not suggest to bring back the former ;-)

I was testing a custom type and in `test/runtests.jl` I defined a method for `isapprox` replacing `==` with `isequal`, so I found a solution for my use case and have no problem.  I'm going to open an ticket to see if other people are interested in this feature.

**Pull request merged**